src/adera/drawing_gl/phong_shader.h

```diff
@@ -110,11 +112,11 @@ inline void sync_drawent_phong(osp::draw::DrawEnt ent, ArgsForSyncDrawEntPhong c
     }
 }
 
-template<typename ITA_T, typename ITB_T>
+template<std::input_iterator IT_T, std::sentinel_for<IT_T> SENT_T>
 void sync_drawent_phong(
-        ITA_T const&                    first,
-        ITB_T const&                    last,
-        ArgsForSyncDrawEntPhong const   args)
+        IT_T const&                   first,
+        SENT_T const&                 last,
+        ArgsForSyncDrawEntPhong const args)
 {
     std::for_each(first, last, [&args] (osp::draw::DrawEnt const ent)
```

Comment:
One of the things I've been considering, is that there is a very low to zero chance that this function will ever be used with sentinels. I don't believe it's worth the extra typing and additional C++ knowledge required to understand its purpose. Regular std::for_each also doesn't support sentinels.